### PR TITLE
[Handlers] Map CharacterSpacing property in LabelHandlers

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Map **CharacterSpacing** property in LabelHandlers.

### Platforms Affected ### 

- iOS
- Android

### PR Checklist ###

- [x] Targets the correct branch
- [x] Include handlers tests
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/UnitExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Xamarin.Platform
+{
+	public static class UnitExtensions
+	{
+		public static float EmCoefficient = 0.0624f;
+
+		public static float ToEm(this double pt)
+		{
+			return (float)pt * EmCoefficient; //Coefficient for converting Pt to Em
```
Comment (hartez) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/UnitExtensions.cs:9]: This ... can't be right. The value for em is dependent on the font size. And even if it weren't, this assumes that we're using something like a 16.025pt font.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/UnitExtensions.cs (hunk 1 above)
Comment (hartez) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/UnitExtensions.cs:9]: Looks like this came straight out of our existing code; I'll open up an issue for it, and we'll deal with it separately.

Comment (hartez) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/UnitExtensions.cs:9]: https://github.com/xamarin/Xamarin.Forms/issues/13859

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LabelExtensions.cs [hunk 2]
```diff
@@ -20,7 +20,15 @@ public static void UpdateTextColor(this TextView textView, ILabel label, Forms.C
 			else
 			{
 				textView.SetTextColor(textColor.ToNative());
-			}				
+			}
+		}
+
+		public static void UpdateCharacterSpacing(this TextView textView, ILabel label)
+		{
+			if (NativeVersion.IsAtLeast(21))
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LabelExtensions.cs]: Right!. Removed the unnecessary API Level check.

**Review by hartez — APPROVED** [review]

File: src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	public static class CharacterSpacingExtensions
+	{
+		public static NSMutableAttributedString? AddCharacterSpacing(this NSAttributedString attributedString, string text, double characterSpacing)
+		{
+			if (attributedString == null && characterSpacing == 0)
+				return null;
+
+			NSMutableAttributedString? mutableAttributedString = attributedString as NSMutableAttributedString;
+
+			if (attributedString == null || attributedString.Length == 0)
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs:15]: Should this be checking `mutableAttributedString ` instead? If not, then what is the logic above for? We could just have a variable definition. But this check also does not seem to fit.

File: src/Core/src/Platform/iOS/LabelExtensions.cs [hunk 4]
```diff
@@ -23,5 +23,16 @@ public static void UpdateTextColor(this UILabel nativeLabel, ILabel label)
 				nativeLabel.TextColor = textColor.ToNative(textColor);
 			}
 		}
+
+		public static void UpdateCharacterSpacing(this UILabel nativeLabel, ILabel label)
+		{
+			if (string.IsNullOrEmpty(label.Text))
+				return;
+
+			var textAttr = nativeLabel.AttributedText?.AddCharacterSpacing(label.Text, label.CharacterSpacing);
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/LabelExtensions.cs:32]: This logic with the attributed text is very different to the decorations PR. That one check for things, this just uses text. Will this override decorations?

File: src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		{
+			if (!string.IsNullOrEmpty(text))
+			{
+				if (characterSpacing == 0 && !mutableAttributedString.HasCharacterAdjustment())
+					return;
+
+				mutableAttributedString.AddAttribute
+				(
+					UIStringAttributeKey.KerningAdjustment,
+					NSObject.FromObject(characterSpacing), new NSRange(0, text != null ? text.Length - 1 : 0)
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs:53]: If this is based on text length, what happens when the text changes?

**Comment by jsuarezruiz** [comment]
Moved to https://github.com/dotnet/maui/pull/463
Let's continue the review and discussions there!

File: src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs (hunk 5 above)
Comment (jsuarezruiz) [reply src/Core/src/Platform/iOS/CharacterSpacingExtensions.cs:53]: Like in the Renderer, in the Handler need to invoke again the CharacterSpacing logic.

**Pull request closed without merging**